SYCL/GroupAlgorithm/SYCL2020/sort.cpp

```diff
@@ -0,0 +1,361 @@
+              << std::endl;
+  q.submit([&](sycl::handler &h) {
+     auto aI1 = sycl::accessor(bufI1, h);
+     sycl::accessor<std::uint8_t, 1, sycl::access_mode::read_write,
+                    sycl::access::target::local>
+         scratch({local_memory_size}, h);
+
+     h.parallel_for<sort_over_group_kernel_name<T, Compare>>(
+         sycl::nd_range<1>(local_range, local_range), [=](sycl::nd_item<1> id) {
+           scratch[0] = std::uint8_t{};
+           auto local_id = id.get_local_id();
+           switch (test_case) {
+           case 0:
+             if constexpr (std::is_same_v<Compare, std::less<T>> &&
+                           !std::is_same_v<T, CustomType>)
+               aI1[local_id] = my_sycl::sort_over_group(
+                   my_sycl::experimental::group_with_scratchpad(
+                       id.get_group(),
+                       sycl::span{&scratch[0], local_memory_size}),
+                   aI1[local_id]);
+             break;
+           case 1:
+             aI1[local_id] = my_sycl::sort_over_group(
+                 my_sycl::experimental::group_with_scratchpad(
+                     id.get_group(),
+                     sycl::span{&scratch[0], local_memory_size}),
+                 aI1[local_id], comp);
+             break;
+           case 2:
+             aI1[local_id] = my_sycl::sort_over_group(
+                 id.get_group(), aI1[local_id],
+                 my_sycl::experimental::default_sorter<Compare>(
+                     sycl::span{&scratch[0], local_memory_size}));
+             break;
+           }
+         });
+   }).wait_and_throw();
+  return 1;
+}
+
+template <typename T, typename Compare>
+int test_joint_sort(sycl::queue &q, std::size_t n_items, std::size_t local,
+                    sycl::buffer<T> &bufI1, Compare comp, int test_case) {
+  auto n = bufI1.size();
+  auto n_groups = (n - 1) / n_items + 1;
+
+  std::size_t local_memory_size =
+      my_sycl::experimental::default_sorter<>::memory_required<T>(
+          sycl::memory_scope::work_group, n);
+  if (local_memory_size >
+      q.get_device().template get_info<sycl::info::device::local_mem_size>())
+    std::cout << "local_memory_size = " << local_memory_size << ", available = "
+              << q.get_device()
+                     .template get_info<sycl::info::device::local_mem_size>()
+              << std::endl;
+  q.submit([&](sycl::handler &h) {
+     auto aI1 = sycl::accessor(bufI1, h);
+     sycl::accessor<std::uint8_t, 1, sycl::access_mode::read_write,
+                    sycl::access::target::local>
+         scratch({local_memory_size}, h);
+
+     h.parallel_for<joint_sort_kernel_name<T, Compare>>(
+         sycl::nd_range<1>{{n_groups * local}, {local}},
+         [=](sycl::nd_item<1> id) {
+           auto group_id = id.get_group(0);
+           auto ptr_keys = &aI1[group_id * n_items];
+           // auto ptr = aI.get_pointer() + group_id * n_items;
```

Comment:
Is this left over from a previous version?

Reply:
Sorry, which version do you mean by previous?

Reply:
Sorry for being confusing.  What I'm really asking is: Why is this comment here?

Reply:
Ah, I see. The comment is to show that the `get_pointer` way also works. I'll add a comment about it